src/app.ts

```diff
@@ -0,0 +1,171 @@
+class Book {
+    constructor(protected title: string, protected author: string, protected genre: string, protected price: number) {
+    }
+
+    get Title(): string {
+        return this.title;
+    }
+    set Title(newTitle: string) {
+        this.title = newTitle;
+    }
+
+    get Author(): string {
+        return this.author;
+    }
+    set Author(newAuthor: string) {
+        this.author = newAuthor;
+    }
+
+    get Genre(): string {
+        return this.genre;
+    }
+    set Genre(newGenre: string) {
+        this.genre = newGenre;
+    }
+
+    get Price(): number {
+        return this.price;
+    }
+    set Price(newPrice: number) {
+        this.price = newPrice;
+    }
+
+    displayDetails(): void {
+        console.log(`Title: ${this.title}`);
+        console.log(`Author: ${this.author}`);
+        console.log(`Genre: ${this.genre}`);
+        console.log(`Price: $${this.price}`);
+    }
+}
+
+class User {
+
+    constructor(private name: string, private email: string, private address: string) {
+    }
+
+    get Name(): string {
+        return this.name;
+    }
+    set Name(newName: string) {
+        this.name = newName;
+    }
+
+    get Email(): string {
+        return this.email;
+    }
+    set Email(newEmail: string) {
+        this.email = newEmail;
+    }
+
+    get Address(): string {
+        return this.address;
+    }
+    set Address(newAddress: string) {
+        this.address = newAddress;
+    }
+}
+
+class Ebook extends Book {
+
+    constructor(title: string, author: string, genre: string, price: number, private format: string) {
+        super(title, author, genre, price);
+    }
+
+    get Format(): string {
+        return this.format;
+    }
+
+    set Format(newFormat: string) {
+        this.format = newFormat;
+    }
+}
+
+abstract class Payment {
+    constructor(protected amount: number) {
```

Comment:
we can avoid passing amount into constructor because you are not using this variable any other place.